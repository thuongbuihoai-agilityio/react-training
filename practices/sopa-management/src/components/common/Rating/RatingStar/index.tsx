import { memo } from 'react';

// Components
import {
  Icon,
  IconType,
  Text
} from '@components/common';

interface RatingStart {
  value?: string;
  type?: string;
  className?: string;
  classNameStar?: string;
}

const RatingStar = ({
  className,
  classNameStar,
  value,
  type
}: RatingStart) => (
  <div className={`${className}`}>
    <div data-testid='rating-star' className={`${classNameStar}`}>
      <Icon iconName={IconType.star} />
      <Icon iconName={IconType.star} />
      <Icon iconName={IconType.star} />
      <Icon iconName={IconType.star} />
      <Icon iconName={IconType.star} />
    </div>
    <Text
      text={value}
      type={type}
    />
  </div>
);

export default memo(RatingStar);
