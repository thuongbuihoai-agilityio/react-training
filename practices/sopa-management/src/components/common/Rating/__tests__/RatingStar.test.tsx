// Libs
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

// Components
import RatingStar from '../RatingStar';

describe('RatingStar component', () => {
  test('should render RatingStar component with type is default', () => {
    const { getByTestId } = render(<RatingStar />);
    expect(getByTestId('rating-star')).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const { container } = render(<RatingStar />);
    expect(container).toMatchSnapshot();
  });
});
