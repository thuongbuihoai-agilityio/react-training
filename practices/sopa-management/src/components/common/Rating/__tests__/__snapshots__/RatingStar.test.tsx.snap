// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RatingStar component matches snapshot 1`] = `
<div>
  <div
    class="undefined"
  >
    <div
      class="undefined"
      data-testid="rating-star"
    >
      <div
        class="undefined icon-star"
        data-testid="icon"
      />
      <div
        class="undefined icon-star"
        data-testid="icon"
      />
      <div
        class="undefined icon-star"
        data-testid="icon"
      />
      <div
        class="undefined icon-star"
        data-testid="icon"
      />
      <div
        class="undefined icon-star"
        data-testid="icon"
      />
    </div>
    <p
      class=" text-"
      data-testid="text"
    >
      Sopa
    </p>
  </div>
</div>
`;
